File: packages/php/src/Types/Customer.php
<?php

declare(strict_types=1);

namespace Ecollect\Types;

/**
 * Customer / payer information.
 */
class Customer
{
    /** @var string */
    public $email;

    /** @var string */
    public $fullName;

    /** @var string|null */
    public $documentType;

    /** @var string|null */
    public $documentNumber;

    /** @var string|null */
    public $phone;

    /** @var string|null */
    public $mobileCountryCode;

    /** @var string|null */
    public $mobileNumber;

    /**
     * @param array<string, mixed> $data
     */
    public function __construct(array $data = [])
    {
        $this->email             = $data['email']               ?? '';
        $this->fullName          = $data['full_name']           ?? '';
        $this->documentType      = $data['document_type']       ?? null;
        $this->documentNumber    = $data['document_number']     ?? null;
        $this->phone             = $data['phone']               ?? null;
        $this->mobileCountryCode = $data['mobile_country_code'] ?? null;
        $this->mobileNumber      = $data['mobile_number']       ?? null;
    }
}

File: packages/swift/Sources/EcollectSDK/Types/Customer.swift
import Foundation

/// Supported document/ID types by country.
public enum DocumentType: String {
    // Colombia
    case cc = "CC"
    case nit = "NIT"
    case passport = "PP"
    case foreignCard = "CE"
    case foreignDoc = "DE"
    // República Dominicana
    case cedula = "CI"
    case rnc = "RNC"
    // México
    case curp = "CURP"
    case ife = "IFE"
    case rfc = "RFC"
}

/// Customer information used for payment and tokenization.
public struct Customer {
    public var firstName: String
    public var lastName: String
    public var email: String
    public var phone: String
    public var documentType: DocumentType
    public var documentNumber: String
    /// Mobile country code (e.g. 57=Colombia, 52=Mexico, 1=Dominican Republic).
    public var mobileCountryCode: String?

    public init(
        firstName: String,
        lastName: String,
        email: String,
        phone: String,
        documentType: DocumentType,
        documentNumber: String,
        mobileCountryCode: String? = nil
    ) {
        self.firstName = firstName
        self.lastName = lastName
        self.email = email
        self.phone = phone
        self.documentType = documentType
        self.documentNumber = documentNumber
        self.mobileCountryCode = mobileCountryCode
    }

    public var fullName: String { "\(firstName) \(lastName)" }
}
